src/axelar/AxelarTransceiver.sol

```diff
@@ -185,7 +185,10 @@ contract AxelarTransceiver is IAxelarTransceiver, AxelarGMPExecutable, Transceiv
     ) internal virtual override {
         AxelarTransceiverStorage storage slot = _storage();
         uint16 sourceChainId = slot.axelarChainIdToId[sourceChain];
-        if (sourceChainId == 0 || slot.transceiverAddressToId[sourceAddress] != sourceChainId) {
+        if (
+            sourceChainId == 0
+                || slot.idToTransceiverAddressHash[sourceChainId] != keccak256(bytes(sourceAddress))
```

Comment:
Why do we need to use a hash for values of this mapping? Message delivery will be pretty expensive if we need to compute a hash every time

Reply:
The transceiver address hex string is more than 32 bytes long. So loading it as a string value requires loading multiple storage slots, which is more expensive than hashing the string

Reply:
Ah yes fair point!

Reply:
it is cheaper than looking up the entire address as a string, so this is the cheapest safe way to do it. (mappings of strings compute the hash of them every time too)